src/testdir/test_float_func.vim

```diff
@@ -174,6 +197,10 @@ func Test_str2float()
   call assert_equal('inf', string(str2float('Inf')))
   call assert_equal('nan', string(str2float('nan')))
   call assert_equal('nan', string(str2float('NaN')))
```

Comment:
I now see that we'd also need to test this:
```
  call assert_equal('inf', string(str2float('+inf')
  call assert_equal('0.0', string(str2float('+0.0')))
  call assert_equal('-0.0', string(str2float('-0.0')))

I suspect that str2float('+inf') will actually give 0.0 instead of 'inf' in Windows, glancing at 8.0.0158 fix.
I'll update the pull request later today with these additional cases.